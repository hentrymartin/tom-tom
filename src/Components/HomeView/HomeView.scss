@import '../../Styles/variables';

.home-view-wrapper {
  display: flex;
  margin-top: 100px;
  justify-content: center;
  a {
    padding: 15px;
    margin-right: 40px;
    border: 1px solid #c1c1c1;
    box-shadow: $box-shadow;
    transition: transform 0.5s;
    text-decoration: none;
    text-transform: uppercase;
    color: #191919;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    &:hover {
      transform: scale(1.1);
    }
    img {
      height: 60px;
    }

    div {
      margin-top: 15px;
    }
  }
}