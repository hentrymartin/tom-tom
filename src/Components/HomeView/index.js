import React from 'react';
import {Link} from 'react-router-dom';
import { Images } from '../../Images';
import './HomeView.scss';

const HomeView = () => ((
  <div className="home-view-wrapper">
    <Link to="/map">
      <img src={Images.MAP_VIEW} alt="Map view icon" />
      <div>Map View</div>
    </Link>

    <Link to="/list">
      <img src={Images.LIST_VIEW} alt="List view icon" />
      <div>List View</div>
    </Link>
  </div>
));

export default HomeView;