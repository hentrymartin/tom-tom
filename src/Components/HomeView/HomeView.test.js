import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import HomeView from './index';

describe('HomeView', () => {
  it('should render correctly', () => {
    const component = shallow(<HomeView />);

    expect(component).toMatchSnapshot();
  });

  it('Check for menus', () => {
    const component = mount(
      <MemoryRouter>
        <HomeView />
      </MemoryRouter>,
    );

    expect(component.find('img').length).toBe(2);
  });
});
