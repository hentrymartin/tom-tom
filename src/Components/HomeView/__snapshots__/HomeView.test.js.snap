// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView should render correctly 1`] = `ShallowWrapper {}`;
